public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -1,339 +1,63 @@
-import React, { PureComponent } from 'react';
-import { InlineFormLabel, LegacyForms, Button } from '@grafana/ui';
-const { Select, Input } = LegacyForms;
-import {
-  AppEvents,
-  SelectableValue,
-  DataSourcePluginOptionsEditorProps,
-  onUpdateDatasourceJsonDataOptionSelect,
-  onUpdateDatasourceResetOption,
-  onUpdateDatasourceJsonDataOption,
-  onUpdateDatasourceSecureJsonDataOption,
-} from '@grafana/data';
+import React, { FC, useEffect, useState } from 'react';
+import { Input, InlineField } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps, onUpdateDatasourceJsonDataOption } from '@grafana/data';
+import { ConnectionConfig } from '@grafana/aws-sdk';
+
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
-import { CloudWatchDatasource } from '../datasource';
-import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
-import { CancelablePromise, makePromiseCancelable } from 'app/core/utils/CancelablePromise';
-import { appEvents } from 'app/core/core';
+import { store } from 'app/store/store';
+import { notifyApp } from 'app/core/actions';
+import { createWarningNotification } from 'app/core/copy/appNotification';
 
-const authProviderOptions = [
-  { label: 'AWS SDK Default', value: 'default' },
-  { label: 'Access & secret key', value: 'keys' },
-  { label: 'Credentials file', value: 'credentials' },
-] as SelectableValue[];
+import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
+import { CloudWatchDatasource } from '../datasource';
 
 export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
-export interface State {
-  regions: SelectableValue[];
-}
+export const ConfigEditor: FC<Props> = (props: Props) => {
```

Comment:
So not sure right now how cloud watch does things, but we have sometimes issues that the config page does not actually saves the default values. Like if you have a default in some select but don't change it, then the json data is not updated and after saving that value is undefined. Usually the data source code than has to supply some default for undefined values but it can be confusing if that changes, which I think could happen now with the auth (ie some people not changing the auth from default value, having nothing in the json data and now with config change I assume that can be different).

Which by the way I wonder if there shouldn't be some change on the backend to handle that if we do not initialise the default data here.

Reply:
Okay, haven't noticed that but I'll look into it. Thanks for the info!